lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1694,23 +1694,51 @@ static LogicalResult verifyPortSymbolUses(FModuleLike module,
   return success();
 }
 
+static LogicalResult verifyEnabledLayers(FModuleLike module,
+                                         SymbolTableCollection &symbolTable) {
+  auto circuitOp = module->getParentOfType<CircuitOp>();
+  for (auto layer : module.getLayers()) {
+    auto *op =
+        symbolTable.lookupSymbolIn(circuitOp, cast<SymbolRefAttr>(layer));
+    if (!op)
+      return module->emitOpError() << "enables unknown layer '" << layer << "'";
+    if (!isa<LayerOp>(op))
+      return module->emitOpError()
+          .append("enables non-layer '", layer, "'")
+          .attachNote(op->getLoc())
+          .append("expected layer");
+  }
+
+  // Check public modules don't enable circuit-disabled layers.
+  if (auto disabledLayers = circuitOp.getDisableLayersAttr()) {
+    auto disabledLayersSymRefs = disabledLayers.getAsRange<SymbolRefAttr>();
+    if (module.isPublic()) {
+      for (auto layer : module.getLayers())
+        if (llvm::is_contained(disabledLayersSymRefs, layer))
+          return module->emitOpError(
+                     "public module has circuit-disabled layer ")
```

Comment:
Is set membership sufficient, or do we need to walk and check if parent layer disabled....?

Reply:
You're going to have to check if a parent layer is disabled

Reply:
Great, thanks!